src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -129,7 +126,7 @@ public function load($class, string $type = null)
         return $collection;
     }
 
-    protected function addRoute(RouteCollection $collection, $annot, $globals, \ReflectionClass $class, \ReflectionMethod $method)
+    protected function addRoute(RouteCollection $collection, $annot, array $globals, \ReflectionClass $class, \ReflectionMethod $method)
```

Comment:
no type for $annot?

Reply:
Yeah, about that…

If I read the code correctly, you can basically build your own annotation class that duck-types most of the methods of our implementation and change this property:

https://github.com/symfony/symfony/blob/940eabb1219a54fdd0237e568b44f009e86d5a13/src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php#L63

So we could declare `object $annot` here, but that's it. If we want to get rid of the duck-typing, we could introduce an interface in 4.4 that the annotation class has to implement.

Reply:
let's add `@param Route $annot or an object that exposes a similar interface`?

Reply:
Here you go: #33234